crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -1443,6 +1485,12 @@ pub enum SuiTransactionKind {
     TransferObject(SuiTransferObject),
     /// Pay one or more recipients from a set of input coins
     Pay(SuiPay),
+    /// Pay one or more recipients from a set of Sui coins, the input coins
+    /// are also used to for gas payments.
+    PaySui(SuiPaySui),
+    /// Pay one or more recipients from a set of Sui coins, the input coins
+    /// are also used to for gas payments.
+    PayAllSui(SuiPayAllSui),
```

Comment:
Will we also eliminate `TransferSui` now?

Reply:
@sblackshear yes I want to if that makes sense, as semantically paySui and payAllSui will cover what TransferSui does today. 
To deprecate that, the steps are after this PR are:
1. implement corresponding changes of paySui and payAllSui in CLI RPC and SDK
2. change callsites of TransferSui to use paySui / payAllSui instead in RPC SDK and CLI
3. deprecate TransferSui native txn type